packages/analytics-node/src/config.ts

```diff
@@ -0,0 +1,40 @@
+import { NodeOptions, NodeConfig as INodeConfig, Event } from '@amplitude/analytics-types';
+import { Config, Http, MemoryStorage, UUID } from '@amplitude/analytics-core';
+
+export class NodeConfig extends Config implements INodeConfig {
+  constructor(apiKey: string, userId?: string, options?: NodeOptions) {
+    const storageProvider = createEventsStorage();
+    const transportProvider = createTransport();
+
+    super({
+      ...options,
+      apiKey,
+      storageProvider,
+      transportProvider,
+      userId: userId,
+      deviceId: createDeviceId(),
+      sessionId: createSessionId(),
+    });
+  }
+}
+
+export const useNodeConfig = (apiKey: string, userId?: string, overrides?: NodeOptions): INodeConfig => {
+  return new NodeConfig(apiKey, userId, overrides);
+};
+
+export const createEventsStorage = () => {
+  const eventsStorage = new MemoryStorage<Event[]>();
+  return eventsStorage;
+};
+
+export const createDeviceId = () => {
+  return UUID();
+};
```

Comment:
We can do a conditional check and if device id is not provided, then generate it.

Reply:
probably not ideal since either we generate one for all events from different user/device combo, or a unique one for each event from the same user/device combo